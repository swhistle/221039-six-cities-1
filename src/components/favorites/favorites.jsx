import React from "react";
import PropsTypes from "prop-types";
import {PlacesListComponent} from "../places-list/places-list.jsx";

export class FavoritesComponent extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {favoriteOffers, addToBookmarks} = this.props;

    if (favoriteOffers) {
      return <main className="page__main page__main--favorites">
        <div className="page__favorites-container container">
          {
            favoriteOffers.length > 0 ? <section className="favorites">
              <h1 className="favorites__title">Saved listing</h1>
              <div className="favorites__list">
                <PlacesListComponent rentObjects={favoriteOffers} favoriteOffersList={favoriteOffers} addToBookmarks={addToBookmarks}/>
              </div>
            </section> : <section className="favorites favorites--empty">
              <h1 className="visually-hidden">Favorites (empty)</h1>
              <div className="favorites__status-wrapper">
                <b className="favorites__status">Nothing yet saved.</b>
                <p className="favorites__status-description">Save properties to narrow down search or plan yor future trips.</p>
              </div>
            </section>
          }
        </div>
      </main>;
    }

    return null;
  }
}

FavoritesComponent.propTypes = {
  favoriteOffers: PropsTypes.array,
  addToBookmarks: PropsTypes.func,
};
