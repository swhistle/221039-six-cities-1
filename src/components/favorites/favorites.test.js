import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from "react-router-dom";
import browserHistory from "../../history";

import {FavoritesComponent} from "./favorites.jsx";

const TEST_FAVORITES_OFFERS = [
  {
    id: 1,
    type: `room`,
    title: `Wood and stone place`,
    price: 80,
    rating: 4,
    location: {
      longitude: 1,
      latitude: 1,
      zoom: 1
    }
  }
];

it(`FavoritesComponent correct renders`, () => {
  const favoritesList = renderer
    .create(
        <Router history={browserHistory}>
          <FavoritesComponent favoriteOffers={TEST_FAVORITES_OFFERS}/>
        </Router>
    ).toJSON();
  expect(favoritesList).toMatchSnapshot();
});
