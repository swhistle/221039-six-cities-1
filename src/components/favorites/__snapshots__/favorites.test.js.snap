// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesComponent correct renders 1`] = `
<main
  className="page__main page__main--favorites"
>
  <div
    className="page__favorites-container container"
  >
    <section
      className="favorites"
    >
      <h1
        className="favorites__title"
      >
        Saved listing
      </h1>
      <div
        className="favorites__list"
      >
        <article
          className="near-places__card place-card"
        >
          <div
            className="near-places__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  80
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                onClick={[Function]}
                title="add to favorites"
                type="button"
              >
                <svg
                  className="property__bookmark-icon"
                  height={19}
                  viewBox="0 0 17 18"
                  width={18}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/1"
                onClick={[Function]}
              >
                Wood and stone place
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              room
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
